charts/k8ssandra/templates/traefik.ingressroutes.yaml

```diff
@@ -43,3 +45,92 @@ spec:
         - name: {{ $clusterName }}-{{ $datacenterName }}-service
           port: 9042
 {{- end }}
+{{- if $traefik.stargate.enabled }}
+---
+{{- if and $traefik.stargate.cassandra.enabled $traefik.cassandra.enabled }}
+  {{- fail "ingress.traefik.stargate.cassandra.enabled and ingress.traefik.cassandra.enabled cannot both be enabled." }}
```

Comment:
The error message here isn't necessarily true. It's possible to enable both, they would just each need their own external port. To be clear users do not have to use 9042 for this traffic, it's just an adjustment client-side to the AddressTranslator used in the driver or SniProxy.

With that being said I struggle to understand a deployment where you would want to expose both the Stargate CQL service _and_ the main cluster's native protocol port. At this point I don't really have a recommendation, just want to comment on the situation and kick off a convo _IF_ we want to change something.

Reply:
@jsanda feel free to chime in here.

Reply:
Sorry, I have not gone through the PR yet :(

>With that being said I struggle to understand a deployment where you would want to expose both the Stargate CQL service and the main cluster's native protocol port. 

@adejanovski didn't you do some benchmarking with Stargate? I would like to hear your thoughts on this.

Reply:
I agree with @bradfordcp, if you deploy a cluster with Stargate then you probably meant to use it as a coordination layer (whatever the protocol) and don't need to access the data nodes native port.
It would be good IMO to take an opinionated approach: either you deploy Stargate nodes and they are your only access points to the cluster or you don't and then have to go through the Cassandra nodes.

Reply:
I am in agreement with @bradfordcp and @adejanovski.

Reply:
Thanks all -- my original thinking was:

1. I'm not aware of a reason to expose both at the same time
2. We don't currently have any precedent for customizing port numbers anywhere in k8ssandra.

I wrote the error message based on what is currently supported by the charts as written, not intending to imply that simultaneous support was technically impossible should we decide we wanted it.

Reply:
I think we just agreed that it's fine as-is. Right? Anyone requesting changes here?

Reply:
@jakerobb it sounds like we are all on the same page and the check enforces what we want. I think we are good to go here 👍